### DISCUSSION TITLE
ModuleNotFoundError: No module named 'example' while running sample add program

### DISCUSSION THREAD

**Discussion by karimkhanp** [pybind/pybind11#4399] [Q&A]
I want to integrate C++ with python. I closely followed the documentation here - https://pybind11.readthedocs.io/en/latest/basics.html

My OS - Mac

Steps to reproduce the error - 
```
$ git clone https://github.com/pybind/pybind11.git
$ cd pybind11
$ mkdir build
$ cd build
$ cmake --version
cmake version 3.25.0

CMake suite maintained and supported by Kitware (kitware.com/cmake).
$ cmake ..
$ make check -j 4
$ cd .. 
(put example.cpp in pybind11 directory)
$ c++ -O3 -Wall -shared -std=c++11 -undefined dynamic_lookup $(python3 -m pybind11 --includes) example.cpp -o example$(python3-config --extension-suffix)
$ python
$ >>> import example
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'example'
```

additional info 
```
(vi_ai) karimkhan@192 pybind11 % whereis python3
python3: /usr/bin/python3 /usr/local/share/man/man1/python3.1
(vi_ai) karimkhan@192 pybind11 % whereis python 
python: /Users/karimkhan/Documents/virtualenv_dir/vi_ai/bin/python
```
So I tried with this command

`c++ -O3 -Wall -shared -std=c++11 -undefined dynamic_lookup $(python -m pybind11 --includes) example.cpp -o example$(python3-config --extension-suffix)`

But still it gives same error. 

I am currently using my virtualenv vi_ai. And while compilation, it usage same virtualenv path so everything looks fine as far as python version concern I believe.

When I do 
```

whereis python3-config
python3-config: /usr/local/bin/python3-config
```

`python3-config` path appears from outside virtualenv path. I don't see `python3-config` in virtualenv site packages. 

Any recommendation? 

I tried same steps on Linux as well but there also getting the same error.

**Comment by LukeLIN-web** [comment]
I follow this https://www.youtube.com/watch?v=H2wOlriHGmM and it works
